Explorer/Assets/DCL/SDKComponents/SceneUI/Tests/UITransformSortingSystemShould.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Arch.Core;
+using CRDT;
+using CrdtEcsBridge.Components;
+using DCL.ECSComponents;
+using DCL.SDKComponents.SceneUI.Components;
+using DCL.SDKComponents.SceneUI.Systems.UITransform;
+using ECS.TestSuite;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.UIElements;
+
+namespace DCL.SDKComponents.SceneUI.Tests
+{
+    public class UITransformSortingSystemShould : UnitySystemTestBase<UITransformSortingSystem>
+    {
+        private static readonly int[] ORDER = Enumerable.Range(0, 10).ToArray();
+
+        private Dictionary<CRDTEntity, Entity> entitiesMap;
+        private Entity sceneRoot;
+
+        [SetUp]
+        public void SetUp()
+        {
+            CreateEntities();
+
+            system = new UITransformSortingSystem(world);
+        }
+
+        [Test]
+        [Repeat(5)]
+        public void SortChildren()
+        {
+            // Act
+            system.Update(0);
+
+            // Assert
+            VisualElement[] expected = ORDER.Select(i => world.Get<UITransformComponent>(entitiesMap[i]).Transform).ToArray();
+            IEnumerable<VisualElement>? actual = world.Get<UITransformComponent>(sceneRoot).Transform.Children();
+
+            CollectionAssert.AreEqual(expected, actual);
+        }
+
+        private void CreateEntities()
+        {
+            entitiesMap = new Dictionary<CRDTEntity, Entity>();
+
+            var root = new UITransformComponent { Transform = new VisualElement() };
+            entitiesMap[SpecialEntitiesID.SCENE_ROOT_ENTITY] = sceneRoot = world.Create(root);
+
+            // shuffle randomly
+            var rng = new Random();
```

Comment:
Using randomness based on the system clock (the default Random constructor) is a weird choice for unit testing, to keep things consistent I would at least choose a pre-defined list of seeds or just TestCase(seed) with some numbers instead of this.

Anyways, since this is a sorting test, no number combination should make it fail, right?

Reply:
good idea